src/aiq/tool/authenticated_request.py

```diff
@@ -0,0 +1,188 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+
+import httpx
+from pydantic import ValidationError
+
+from aiq.authentication.exceptions.call_back_exceptions import AuthenticationError
+from aiq.authentication.exceptions.request_exceptions import BaseUrlValidationError
+from aiq.authentication.exceptions.request_exceptions import BodyValidationError
+from aiq.authentication.exceptions.request_exceptions import HTTPHeaderValidationError
+from aiq.authentication.exceptions.request_exceptions import HTTPMethodValidationError
+from aiq.authentication.exceptions.request_exceptions import QueryParameterValidationError
+from aiq.authentication.interfaces import AuthProviderBaseConfig
+from aiq.data_models.authentication import AuthResult
+from aiq.data_models.authentication import HTTPMethod
+from aiq.data_models.authentication import HTTPResponse
+from aiq.utils.request_utils import HTTPRequestValidator
+
+logger = logging.getLogger(__name__)
+
+
+async def make_authenticated_request(url: str,
```

Comment:
Were setting ourselves up to need coverage for each and every possible combination of settings that can be made in a request.

What if we took a different approach like what authlib did? Where we derive from `httpx.AsyncClient` and add in our auth pieces? https://github.com/authlib/authlib/blob/main/authlib/integrations/httpx_client/oauth2_client.py#L63